### PR TITLE
Improve error message when attempting to publish without publish script defined

### PR DESCRIPTION
**Pull request by benmccann**
Prior to this PR it would tell you `No changesets found` if you omitted the `publish` script, and it was very hard to figure out what to do. It was also misleading as the only reason there were no changesets present is because they were deleted earlier in the release process

### CODE REVIEW

File: .changeset/breezy-garlics-bathe.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@changesets/action": minor
```
Comment (Andarist) [inline .changeset/breezy-garlics-bathe.md:2]: IMHO the proposed change is a major change. It's a substantial change in the default behavior. If anybody is relying on publish being an opt-in then it could have a devastating effect for them if we suddenly try to publish their packages out of a blue.

Comment (emmatown) [reply .changeset/breezy-garlics-bathe.md:2]: Yeah, like I'm personally using it in places where publishing is handled separately where I definitely wouldn't want this behaviour and the README documents a `Without Publishing` workflow which with this PR isn't accurate

Comment (benmccann) [reply .changeset/breezy-garlics-bathe.md]: Ah, thanks! I'd never used this without publishing, so sorry I missed that was a way to use this. I've updated this PR to revert to the previous behavior but update the error message

File: src/index.ts [hunk 2]
```diff
@@ -48,11 +48,11 @@ const getOptionalInput = (name: string) => core.getInput(name) || undefined;
 
   switch (true) {
     case !hasChangesets && !hasPublishScript:
-      core.info("No changesets found");
+      core.info("No publish script found");
       return;
     case !hasChangesets && hasPublishScript: {
       core.info(
-        "No changesets found, attempting to publish any unpublished packages to npm"
```
Comment (Andarist) [inline src/index.ts:55]: i feel like this was indicting why an attempt to publish was made, so i think i'd revert this change

File: src/index.ts [hunk 3]
```diff
@@ -48,11 +48,11 @@ const getOptionalInput = (name: string) => core.getInput(name) || undefined;
 
   switch (true) {
     case !hasChangesets && !hasPublishScript:
-      core.info("No changesets found");
+      core.info("No publish script found");
```
Comment (Andarist) [inline src/index.ts:51]:
```suggestion
      core.info("No publish script and no changesets found");
```

File: .changeset/breezy-garlics-bathe.md (hunk 1 above)
Comment (Andarist) [reply .changeset/breezy-garlics-bathe.md]: We can take another look if this is he best default behavior but that will have to be done as part of the next major release

File: src/index.ts (hunk 2 above)
Comment (benmccann) [reply src/index.ts:55]: Done

**Review by Yedfree — APPROVED** [review]
